bundles/config/org.eclipse.smarthome.config.core/src/main/java/org/eclipse/smarthome/config/core/internal/ConfigMapper.java

```diff
@@ -159,6 +160,11 @@ private static Object objectConvert(Object value, Class<?> type) {
                 result = Integer.valueOf(bdValue);
             } else if (type.equals(Boolean.class) || typeName.equals("boolean")) {
                 result = Boolean.valueOf(bdValue);
+            } else // Handle enums
+            if (type.isEnum()) {
+                @SuppressWarnings({ "rawtypes" })
+                final Class<? extends Enum> enumType = (Class<? extends Enum>) type;
+                return Enum.valueOf(enumType, value.toString());
```

Comment:
You should handle `IllegalArgumentException`s here.

Reply:
The call to `objectConvert()` is covered by an `Exception` catch:
```
try {
 value = objectConvert(value, type);
 FieldUtils.writeField(configuration, fieldName, value, true);
} catch (Exception ex) {
  logger.warn("Could not set field value for field '{}': {}", fieldName, ex.getMessage(), ex);
}
```

If I catch it locally within  `objectConvert()`, I would not know what to return as a valid value. I could make something up like returning the first enum entry, but I thing it is smarter to let the field be untouched. The user will probably have assigned an enum entry that signals him that something is faulty, like in my example where the initial value is "UNKNOWN".

Reply:
ok, makes sense, thanks!